Model/Product/Builder.php

```diff
@@ -135,6 +142,42 @@ public function build(Product $product, StoreInterface $store)
     }
 
     /**
+     * Helper method to fetch and return the normalised rating value for a product. The rating is
+     * normalised to a 0-5 value.
+     *
+     * @param Product $product the product whose rating value to fetch
+     * @param StoreInterface $store the store scope in which to fetch the rating
+     * @return float the normalized rating value of the product
+     */
+    private function buildRatingValue(Product $product, StoreInterface $store) {
+        /** @noinspection PhpUndefinedMethodInspection */
+        if (!$product->getRatingSummary()) {
+            $this->reviewFactory->create()->getEntitySummary($product, $store->getId());
+        }
+
+        /** @noinspection PhpUndefinedMethodInspection */
+        return round($product->getRatingSummary()->getRatingSummary() / 20, 1);
```

Comment:
I looked at how Magento is doing it in the ReviewRenderer and they weren't doing any null checks. Also, it seems to work fine when the product doesn't have any ratings or reviews.